pkg/bootstrap/platform/gcp.go

```diff
@@ -291,18 +291,24 @@ func zoneToRegion(z string) (string, error) {
 func (e *gcpEnv) Locality() *core.Locality {
 	var l core.Locality
 	if e.shouldFillMetadata() {
-		z, zerr := metadata.Zone()
-		if zerr != nil {
-			log.Warnf("Error fetching GCP zone: %v", zerr)
+		_, _, _, envLoc := parseGCPMetadata()
```

Comment:
can you explain (maybe by comments in code) why we re-parse the env var instead of just always calling e.Metadata()

Reply:
My point was Metadata() already has this logic, why do we need to copy it everywhere? If the env var is set, `Metadata()` will be cheap right?

Reply:
I am not sure why this was resolved?

Reply:
oh I missed your 2nd comment in this thread. I agree Metadata() has the same logic but it is fetching instance details as well which is of no use (adding additional complexity) and it's not feasible to update Metadata() function as it is being called from many places. And, probably we might get issues there. That's why I thought it's better to keep it separate. And, since it's a variable parsing which is not costly imo. wdyt?